test/Concurrency/async_sequence_macosx.swift

```diff
@@ -15,4 +15,14 @@ func f<S: AsyncSequence>(s: S) async throws {
   }
 }
 
-
+// Make sure we don't complain about crossing a concurrency boundary here.
+@MainActor
+class Store<Action: Sendable> {
+    private func intercept(_ action: Action) async {
+        await withTaskGroup(of: Optional<Action>.self) { group in
+            for await case let nextAction? in group {
+                _ = nextAction
+            }
+        }
```

Comment:
```suggestion
        }
        
        try await withThrowingTaskGroup(of: Optional<Action>.self) { group in
             for await case let nextAction? in group {
                 _ = nextAction
             }
         }
```

worth adding here?

Reply:
was done in https://github.com/apple/swift/commit/5a1c26856f3e90927a5c687e959972eb547abb84